src/west/app/project.py

```diff
@@ -1080,7 +1080,21 @@ def ensure_cloned(project, stats, take_stats):
                 start = perf_counter()
 
             log.small_banner(f'{project.name}: initializing')
-            project.git(['init', project.abspath], cwd=util.west_topdir())
+            try:
+                # Silence the verbose and repetitive init.defaultBranch
+                # warning. The branch 'placeholder' will never have any
+                # commit so it will never actually exist.
+                project.git(['init', '--initial-branch=placeholder',
+                             project.abspath],
+                            capture_stdout=True, capture_stderr=True,
+                            cwd=util.west_topdir())
+            except subprocess.CalledProcessError:
+                # If git < 2.28 then the above failed because of
+                # --initial-branch. If git >= 2.28 then it failed for
+                # some other reason and will fail again for the same
+                # reason which will be printed this time:
```

Comment:
Can you rework this to use #494 to only try to use `--initial-branch` if the git version is at least `(2, 28, 0)`? If the version is too old, then just fall back on the existing init command in the current code.

With 2.28, I don't see the need for a try/catch.

Reply:
Will do after #494 is merged.

Reply:
> Will do after #494 is merged.

Can you give it a +1 if it looks good to you? That would help get it merged.

Reply:
> Can you rework this to use #494 to only try to use --initial-branch if the git version is at least (2, 28, 0)

Done.